pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1214,34 +1206,48 @@ func TestNodeOperators(t *testing.T) {
 			logger, ctx := ktesting.NewTestContext(t)
 			ctx, cancel := context.WithCancel(ctx)
 			defer cancel()
-			expected := buildNodeInfo(test.node, test.pods)
-			node := test.node
+			expected := buildNodeInfo(&node1.Node, test.pods)
+			node := test.nodes[node1.GetName()]
 
 			cache := newCache(ctx, time.Second, time.Second)
-			cache.AddNode(logger, node)
+			for _, nodeItem := range test.nodes {
+				cache.AddNode(logger, nodeItem)
+			}
 			for _, pod := range test.pods {
 				if err := cache.AddPod(logger, pod); err != nil {
 					t.Fatal(err)
 				}
 			}
+			for index := range expected.Node().Status.Images {
+				for _, name := range expected.Node().Status.Images[index].Names {
+					if imageState, ok := cache.imageStates[name]; ok {
+						expected.ImageStates[name] = imageState
```

Comment:
avoid automatic population of "expected" values. Instead, make the expectations part of the test table